lib/api/client-commands/saveScreenshot.js

```diff
@@ -20,16 +20,12 @@ class SaveScreenshot extends ClientCommand {
   performAction(callback) {
     const {fileName, client} = this;
 
-    this.api.screenshot(this.api.options.log_screenshot_data, function(result) {
+    this.api.screenshot(this.api.options.log_screenshot_data, async function(result) {
       if (client.transport.isResultSuccess(result) && result.value) {
-        Screenshots.writeScreenshotToFile(fileName, result.value, err => {
-          if (err) {
-            Logger.error(`Couldn't save screenshot to "${fileName}":`);
-            Logger.error(err);
-          }
-
+        const file = await Screenshots.writeScreenshotToFile(fileName, result.value);
+        if (file) {
           callback.call(this, result);
```

Comment:
I think this should be outside the `if` block.

Reply:
In the case where the screenshot is returned but there is an error while writing the screenshot to the file, the callback should be called.

Reply:
>In the case where the screenshot is returned but there is an error while writing the screenshot to the file, the callback should be called.

I think callback is not needed as I have added a `catch` in the `writeScreenshotToFile` function. 
https://github.com/nightwatchjs/nightwatch/blob/92f48e2592d7038bfceff69ec81882cae50667bb/lib/utils/screenshots.js#L44-L46

Reply:
>I think this should be outside the if block.

Yes, `if` is redundant, will remove